# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 997e9ca2e1e9
# Full Hash: 997e9ca2e1e9d636f8f956151db81ca2a8a60c82
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement InitModuleLoader methods for WorkerScriptLoader and WorkerGlobalScope; r=jonco
#   
#   This implements a method to initialize the moduleLoader for workers. This will initialize only once, for all worker types (module and classic).
#   
#   Depends on D147324
# ==============================================================================

diff -r ebeb65b42212 -r 997e9ca2e1e9 dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Wed Jan 18 13:46:29 2023 +0000
+++ b/dom/workers/WorkerScope.h	Wed Jan 18 13:46:29 2023 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_workerscope_h__
 
 #include "js/TypeDecls.h"
+#include "js/loader/ModuleLoaderBase.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/Attributes.h"
 #include "mozilla/DOMEventTargetHelper.h"
@@ -156,6 +157,18 @@
 
   Console* GetConsoleIfExists() const { return mConsole; }
 
+  void InitModuleLoader(JS::loader::ModuleLoaderBase* aModuleLoader) {
+    if (!mModuleLoader) {
+      mModuleLoader = aModuleLoader;
+    }
+  }
+
+  // The nullptr here is not used, but is required to make the override method
+  // have the same signature as other GetModuleLoader methods on globals.
+  JS::loader::ModuleLoaderBase* GetModuleLoader(JSContext* aCx) override {
+    return mModuleLoader;
+  };
+
   uint64_t WindowID() const;
 
   // Usually global scope dies earlier than the WorkerPrivate, but if we see
@@ -180,6 +193,7 @@
 
  private:
   RefPtr<Console> mConsole;
+  RefPtr<JS::loader::ModuleLoaderBase> mModuleLoader;
   const UniquePtr<ClientSource> mClientSource;
   nsCOMPtr<nsISerialEventTarget> mSerialEventTarget;
   bool mShouldResistFingerprinting;